test/e2e/storage/testsuites/multiattach.go

```diff
@@ -0,0 +1,271 @@
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Creating pod with the two volumes with the same volume mode")
+		pod, err := framework.CreateSecPodWithNodeName(l.cs, l.ns.Name,
+			[]*v1.PersistentVolumeClaim{l.resource1.pvc, l.resource2.pvc},
+			false, "", false, false, framework.SELinuxLabel,
+			nil, l.config.ClientNodeName, framework.PodStartTimeout)
+		defer func() {
+			framework.ExpectNoError(framework.DeletePodWithWait(f, l.cs, pod))
+		}()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Showing debug information")
+		framework.Logf("PVC1: %+v", l.resource1.pvc)
+		framework.Logf("PV1: %+v", l.resource1.pv)
+		framework.Logf("PVC2: %+v", l.resource2.pvc)
+		framework.Logf("PV2: %+v", l.resource2.pv)
+		framework.Logf("Pod: %+v", pod)
+
+		utils.VerifyExecInPodSucceed(pod, "ls -l /mnt/")
+		utils.VerifyExecInPodSucceed(pod, "mount")
+		utils.VerifyExecInPodSucceed(pod, "cat /proc/partitions")
+
+		By("Checking if the 1st volume exists as expected volume mode")
+		utils.CheckVolumeModeOfPath(pod, l.resource1.pattern.VolMode, "/mnt/volume1")
+
+		By("Checking if read/write to the 1st volume works properly")
+		utils.CheckReadWriteToPath(pod, l.resource1.pattern.VolMode, "/mnt/volume1")
+
+		By("Checking if the 2nd volume exists as expected volume mode")
+		utils.CheckVolumeModeOfPath(pod, l.resource2.pattern.VolMode, "/mnt/volume2")
+
+		By("Checking if read/write to the 2nd volume works properly")
+		utils.CheckReadWriteToPath(pod, l.resource2.pattern.VolMode, "/mnt/volume2")
+	}
+
+	// This tests below configuration:
+	//          [pod1]
+	//          /    \      <- same volume mode
+	//   [volume1]  [volume2]
+	It("should attach two volumes with the same volume mode to the same pod", func() {
+		init()
+		defer cleanup()
+
+		l.resource2.pattern.VolMode = l.resource1.pattern.VolMode
+		testAttachTwoVolumesToOnePod(l)
+	})
+
+	// This tests below configuration:
+	//          [pod1]
+	//          /    \      <- different volume mode (only <block, filesystem> pattern is tested)
+	//   [volume1]  [volume2]
+	It("should attach two volumes with different volume mode to the same pod", func() {
+		if pattern.VolMode == v1.PersistentVolumeFilesystem {
+			framework.Skipf("Filesystem volume case should be covered by block volume case -- skipping")
+		}
+
+		init()
+		defer cleanup()
+
+		l.resource2.pattern.VolMode = v1.PersistentVolumeBlock
+		testAttachTwoVolumesToOnePod(l)
+	})
+
+	// This tests below configuration:
+	// [pod1] [pod2]
+	// [   node1   ]
+	//   \      /     <- same volume mode
+	//   [volume1]
+	It("should attach the same volume with the same volume mode to the different pod on the same node", func() {
+		init()
+		defer cleanup()
+
+		var err error
+
+		// Decide the node to deploy both 1st pod and 2nd pod.
+		// In this test, pods are depoyed to the node other than the node
+		// that is defined in l.config.ClientNodeName.
```

Comment:
why can't it be ClientNodeName?

Reply:
This intends to deploy both 1st pod and 2nd pod to the same node.

I think that some drivers can properly attach volumes to different nodes at the same time, but others can't.

For example, NFS and CephFS should be able to attach a volume to multiple nodes at the same time and the volume could be consumed by any pods at the same time. However, iSCSI and rbd won't be able to attach a volume to multiple nodes at the same time, especially when it is used as filesystem volume. Because, it will destroy metadata on the filesystem.
(Also, some cloud providers won't allow attaching a volume to multiple nodes, at the same time. And local volume won't be attach to another nodes.)

So, I think that we need to separate tests for the same node case and different node case. And the background of adding this test case was happened in the same node case, I focused this case here. (We could also add different node case.)

Reply:
I started to think that above is just for the case if ClientNodeName hasn't been set, already. If driver requires to deploy on somewhere for some reason, like hostpath driver do, we should use this value.(In that case, we will still be able to make both pod deploy on the same node, without updating the node again here.)

Reply:
OK i see, thanks. I don't think a different node case is needed, maybe later. Yeah, probably ClientNodeName is empty sometimes.